app/models/promotion.rb

```diff
@@ -6,7 +6,7 @@ class Promotion < ActiveRecord::Base
 
   validates :user_id, uniqueness: { scope: :promo_id }
 
-  scope :with_active_promos, ->{ where(promo_id: Promo.active.map(&:id)) }
+  scope :with_active_promos, -> { where(promo_id: Promo.active.select(:id)) }
```

Comment:
or even `pluck`, is there any difference between `select` and `pluck`?

Reply:
With Select doing one query